plugins/telemetry-otel/src/main/java/org/opensearch/telemetry/tracing/sampler/RequestSampler.java

```diff
@@ -44,15 +43,15 @@ public SamplingResult shouldSample(
         Attributes attributes,
         List<LinkData> parentLinks
     ) {
-
         final String trace = attributes.get(AttributeKey.stringKey(TRACE));
 
         if (trace != null) {
             return (Boolean.parseBoolean(trace) == true) ? SamplingResult.recordAndSample() : SamplingResult.drop();
-        } else {
-            return defaultSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
         }
-
+        if (fallbackSampler != null) {
+            return fallbackSampler.shouldSample(parentContext, traceId, name, spanKind, attributes, parentLinks);
+        }
+        return SamplingResult.drop();
```

Comment:
It means we don't have configured any Sampler so shouldn't it be Sample all the requests instead of drop?